laghos.cpp

```diff
@@ -933,6 +934,10 @@ int main(int argc, char *argv[])
         // usual time loop when rom_restore phase is false.
         std::ofstream outfile_tw_steps("run/tw_steps");
         timeLoopTimer.Start();
+        if (rom_hyperreduce && rom_GramSchmidt) 
+        {
+            romOper->InducedGramSchmidtInitialize(romS);
```

Comment:
@siuwuncheung It looks like this `InducedGramSchmidtInitialize` function is called once, right after each `ROM_Operator` constructor call. Could you just put this at the end of the `ROM_Operator` constructor? Also, it looks the `ComputeReducedMv` and `ComputeReducedMe` at the end of the `ROM_Operator` constructor can be removed.

Reply:
@dylan-copeland For your first concern, at the beginning I thought the same and put `InducedGramSchmidtInitialize` right after `ROM_Operator` call as it is more clear. However, `InducedGramSchmidtInitialize` also change the coefficients in `romS` and I think it is better to make sure the coefficients are not unnecessarily affected in the restore phase. Do you think it is more appropriate to put it after the `ROM_Operator` call? For your second concern, practically `ComputeReducedMv` and `ComputeReducedMe` are not needed, but in the current code `rom_GramSchmidt = true` does not turn `rom_reduceMass = false`. I kept (and add) those calls at the end after any updates to the basis. It can be modified easily though if we do not need those anymore.

Reply:
@siuwuncheung This makes sense, as `romS` must be modified. There is some setup in `InducedGramSchmidtInitialize`, i.e. the calls to `InducedGramSchmidtM{v,e}`, which does not involve `romS` and could be in `SetupHyperreduction`, but it is not a big deal. It is unclear whether `rom_reduceMass=true` will ever be used, so maybe we can remove that if everyone agrees. If Gram-Schmidt works well, then I don't see a reason to keep the reduced mass matrix version. @chldkdtn Do you see any reason to keep the reduced mass matrix version (without Gram-Schmidt orthonormalization)?